packages/bloc_lint/lib/src/rules/prefer_context_extensions.dart

```diff
@@ -0,0 +1,43 @@
+// ignore: implementation_imports
+import 'package:_fe_analyzer_shared/src/parser/parser.dart';
+import 'package:bloc_lint/bloc_lint.dart';
+
+/// {@template prefer_context_extensions}
+/// The prefer_context_extensions lint rule.
+/// {@endtemplate}
+class PreferContextExtensions extends LintRule {
+  /// {@macro prefer_context_extensions}
+  PreferContextExtensions([Severity? severity])
+    : super(name: rule, severity: severity ?? Severity.warning);
+
+  /// The name of the lint rule.
+  static const rule = 'prefer_context_extensions';
+
+  @override
+  Listener? create(LintContext context) {
+    return _Listener(context);
+  }
+}
+
+class _Listener extends Listener {
+  _Listener(this.context);
+
+  final LintContext context;
+
+  @override
+  void handleIdentifier(Token token, IdentifierContext _) {
+    if (token.lexeme == 'of') {
+      final prev = token.previous;
+      if (prev != null && prev.lexeme == '.') {
+        final target = prev.previous;
+        if (target != null && target.lexeme == 'BlocProvider') {
```

Comment:
I don't think this is quite right because there are also `RepositoryProvider` extensions that this rule should apply to. In addition, I think we should try to unify some of the logic across this PR and #4572 since they are closely related.

Reply:
I've updated the logic to support `RepositoryProvider` as well. I also reviewed https://github.com/felangel/bloc/pull/4572. In my case, detecting `RepositoryProvider` and `BlocProvider` usage is much simpler when handled separately. Combining the two makes the logic harder to read and understand. So, I believe it's not necessary to unify them, since they serve different purposes.

Reply:
Sounds good will review shortly, thanks for your contributions! 💙